tests/lib/nested.sh

```diff
@@ -1142,9 +1110,6 @@ nested_start_core_vm_unit() {
     # wait for the $NESTED_VM service to appear active
     wait_for_service "$NESTED_VM"
 
-    # make sure the service started and it is running
-    nested_retry_start_with_boot_errors
```

Comment:
I think you should explain a bit better (maybe in the commit message) why this is being removed. It sounds like a good step to have.

If the reason is that it doesn't work on intel machines, maybe you can just modify the nested_retry_start_with_boot_errors() function to have a check on `cpuinfo` and return if it's an Intel?

Reply:
Thanks for the review @mardy 

This function is a bit confusing, when I added that I promised to remove it once the error is gone, with the change that I did in this pr the error is almost not reproduced but it is still happening. 

Now, the error is not happening when we boot the machine, it happens after a while, so this check is not working anymore, I'll change the approach and I'll create a service to monitor the nested-vm and restart it in case this error occurs at any moment. My idea is to create a different pr for that. Does it make sense?

Reply:
I'll try to push this new service in the next commit so we fix the issue now.

Reply:
@mardy I already included a new solution for that. please take a second look